### PR TITLE
Use travis docker features

### PR DESCRIPTION
**Pull request by iamjarvo**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

### Description of the Change

This allows us to run tests against an instance of Vinyl running in a container

### Why Should This Be In The Gem?

Allows us run tests and verify changes

### Benefits

Allows us run tests and verify changes

### Possible Drawbacks

The current implementation is slow

### Verification Process

Test should run with failures. Since its running against a fresh instantiation of vinyl the current test suite will fail. There shouldn't be any connection errors

### Applicable Issues (Optional)

This is a very simple implementation. It makes some assumptions and doesn't do error checking. Future PRs will make it quicker, more robust, and friendlier. The test suite will also eventually be green.

### CODE REVIEW

**Comment by iamjarvo** [comment]
Hmm, need to figure out why travis isn't running for PRs. You can take a look at an example run on my fork https://travis-ci.org/iamjarvo/vinyldns-ruby/builds/474699336

File: scripts/start_api [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+mkdir -p ./api
+git clone https://github.com/vinyldns/vinyldns ./api/
+./api/bin/docker-up-vinyldns.sh
```
Comment (mdb) [inline scripts/start_api:5]: I _believe_ this starts both the portal and the API. You may want to consider `bin/docker-up-vinyldns.sh --api-only`, which will _only_ start the API.

**Comment by britneywright** [comment]
> Hmm, need to figure out why travis isn't running for PRs. You can take a look at an example run on my fork https://travis-ci.org/iamjarvo/vinyldns-ruby/builds/474699336

Try pushing an update, I updated the Travis settings think it should run now.

File: Makefile [hunk 2]
```diff
@@ -1,5 +1,8 @@
-init:
-	pip install -r requirements.txt
+start-api:
+		./scripts/start_api
```
Comment (britneywright) [inline Makefile]: Is there extra spacing here that's not needed?

File: scripts/start_api (hunk 1 above)
Comment (iamjarvo) [reply scripts/start_api:5]: I will add this later. Wanted to do the bare minimum to get this working and explore the options a little. I've also been using the portal locally to see what test data is being generated used.

**Review by britneywright — APPROVED** [review]

**Pull request merged**